actors/evm/src/interpreter/execution.rs

```diff
@@ -181,222 +159,236 @@ impl<'r, 'a, RT: Runtime + 'r> Machine<'r, 'a, RT> {
                 break;
             }
 
-            match self.step()? {
-                ControlFlow::Continue => {
-                    self.pc += 1;
-                }
-                ControlFlow::Jump => {}
-                ControlFlow::Exit => {
-                    break;
-                }
-            };
+            self.step();
+
+            if self.exit.is_some() {
+                break;
+            }
         }
 
-        Ok(())
+        match &self.exit {
+            None => Ok(()),
+            Some(StatusCode::Success) => Ok(()),
+            Some(err) => Err(err.clone()),
+        }
     }
 
-    fn step(&mut self) -> Result<ControlFlow, StatusCode> {
-        let op = OpCode::try_from(self.bytecode[self.pc])?;
-        Self::JMPTABLE[op as usize](self)
+    fn step(&mut self) {
+        let op = OpCode::try_from(self.bytecode[self.pc]);
+        match op {
+            Ok(op) => Self::JMPTABLE[op as usize](self),
+            Err(err) => self.exit = Some(err),
+        }
```

Comment:
Instead of checking if PC is "in bounds above and inside step, we can just check inside step:

```suggestion
        match self.bytecode.get(self.pc).ok_or(StatusCode::Success).map(OpCode::try_from) {
            Ok(op) => Self::JMPTABLE[op as usize](self),
            Err(err) => self.exit = Some(err),
        }
```

Then remove the `self.pc < self.bytecode.len()` check above.

Reply:
We can _probably_ also use unsafe when accessing the jumptable. _That_ I'm happy with given that we _know_ it's safe.

Reply:
I'll try those too.